invenio_accounts/datastore.py

```diff
@@ -37,29 +73,39 @@ def commit(self):
         datastore_post_commit.send(session=self.db.session)
         current_db_change_history.clear_dirty_sets(self.db.session)
 
-    def put(self, model):
```

Comment:
Put was being used for both roles and users, but the ``mark_changed()`` call only handled users, and in addition the model object only have an id if it was committed which is not always the case.